bin/proving-service/src/commands/mod.rs

```diff
@@ -69,18 +67,51 @@ impl Default for ProxyConfig {
 }
 
 impl ProxyConfig {
-    /// Loads config file from current directory and default filename and returns it
+    /// Load config from environment variables using defaults in case of missing values.
     ///
-    /// This function will look for the configuration file with the name defined at the
-    /// [PROVING_SERVICE_CONFIG_FILE_NAME] constant in the current directory.
-    pub(crate) fn load_config_from_file() -> Result<ProxyConfig, String> {
-        let mut current_dir = std::env::current_dir().map_err(|err| err.to_string())?;
-        current_dir.push(PROVING_SERVICE_CONFIG_FILE_NAME);
-        let config_path = current_dir.as_path();
+    /// The environment variables must be prefixed with [`ENV_PREFIX`].
+    pub(crate) fn load() -> Result<ProxyConfig, ProvingServiceError> {
+        let mut config = ProxyConfig::default();
+
+        config.host = std::env::var(format!("{}HOST", ENV_PREFIX)).unwrap_or(config.host);
+        config.port = std::env::var(format!("{}PORT", ENV_PREFIX))
+            .unwrap_or(config.port.to_string())
+            .parse()?;
+        config.timeout_secs = std::env::var(format!("{}TIMEOUT_SECS", ENV_PREFIX))
+            .unwrap_or(config.timeout_secs.to_string())
+            .parse()?;
```

Comment:
nit: we should be able to write the format as: `format!("{ENV_PREFIX}TIMEOUT_SECS")`.

But also, it may be worth to create a small utility function which will improve readability of this section - e.g., to make this line look something like this:
```rust
config.timeout_secs = read_env_u64("TIMEOUT_SECS", config.timeout_secs)?;
```

Reply:
I ended up merging `ProxyConfig` and `StartProxy`, which allowed us to remove the whole method. It not uses `clap` directly to initialize the struct.